shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -1137,4 +1233,55 @@ public void release() {
             mNativeView.getFlutterJNI().unregisterTexture(id);
         }
     }
+
+    private class PointerState {
+        private long device;
+        private double dx;
+        private double dy;
+        private boolean pressed;
+
+        public PointerState(long device, double dx, double dy) {
+            this.device = device;
+            this.dx = dx;
+            this.dy = dy;
+            this.pressed = false;
+        }
+
+        public boolean isLastLocation(double dx, double dy) {
+            return this.dx == dx && this.dy == dy;
+        }
+
+        public void setLastLocation (double dx, double dy) {
+            this.dx = dx;
+            this.dy = dy;
+        }
+
+        public boolean isDown() {
```

Comment:
should `pressed` just be non-private instead of having this getter and the two setter?